packages/form/src/components/CheckboxGroupField/index.tsx

```diff
@@ -38,14 +56,47 @@ export const CheckboxGroupField = <
   name,
   required = false,
   shouldUnregister = false,
+  rules,
 }: CheckboxGroupFieldProps<TFieldValues, TName>) => {
   const { getError } = useErrors()
+  const validate = useCallback(
+    (value: string[]) => {
+      const requiredChildren =
+        Children.map(children, child => {
+          if (isValidElement<ElementProps>(child)) {
+            if (child.props.required) {
+              return child.props.name
+            }
+
+            return null
+          }
+
+          return null
+        })?.filter(Boolean) ?? []
+
+      if (!required && arraysContainSameValues(value, requiredChildren)) {
+        return undefined
+      }
+
+      if (value.length >= Children.count(children)) {
```

Comment:
With 2 not required checkbox and 1 required checkbox.
If we check the 2 not required checkbox, the field gonna be valid.

Reply:
It should work just fine this condition is invalid in that case too as you have only 2 checkboxes out of 3 checked


https://github.com/scaleway/ultraviolet/assets/15812968/df239efe-4c3e-4352-8527-97b5e73d67a1